cstar/roms/runtime_settings.py

```diff
@@ -0,0 +1,806 @@
+from os import PathLike
+from pathlib import Path
+from typing import ClassVar, Optional, Union, get_args, get_origin
+
+from pydantic import (
+    BaseModel,
+    Field,
+    ModelWrapValidatorHandler,
+    model_serializer,
+    model_validator,
+)
+
+from cstar.base.log import get_logger
+from cstar.base.utils import _list_to_concise_str
+
+log = get_logger(__name__)
+
+################################################################################
+# Formatting methods for serializer:
+
+
+def _format_list_of_floats(float_list: list[float]) -> str:
+    joiner = " "
+    return joiner.join(_format_float(x) for x in float_list)
+
+
+def _format_list_of_paths(
+    path_list: list[Path], multi_line: Optional[bool] = False
+) -> str:
+    joiner = "\n    " if multi_line else " "
```

Comment:
1. missing docstring
2. the delimiter `"\n   "` is duplicated and should be placed into a constant to avoid requirement to change in two places later and to keep the spacing consistent (e.g. typos if adding another copy somewhere else later).